models/nlp/albert/run_pretraining.py

```diff
@@ -407,6 +403,13 @@ def main():
     global max_grad_norm
     max_grad_norm = train_args.max_grad_norm
 
+    # TODO : Change to obfuscate smddpcommon. This code does not use GradientTape, so need to pass it like this.
+    if train_args.bucket_cap_mb:
```

Comment:
A side question here, why we need explicit call hc.setBucketSize()?
Is there any issue let herring allocate the bucket memory and use env var to control the size?

Reply:
I guess it's a API decision to control it via python API rather than env vars. The env var `HERRING_BUCKET_SIZE` is for us to easily debug, it still has `HERRING` name in it.`bucket_cap_mb` is more like a hyper parameter during sagemaker training rather than an extra env var.

For PyTorch it's easy, specify hyper param in training config, it will pass to training script, then pass to DDP class init.  
https://github.com/aws/herring/blob/master/smdistributed/dataparallel/torch/parallel/distributed.py#L54
https://github.com/HerringForks/DeepLearningExamples/blob/master/PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py#L118

For TF there are 2 ways, if the training script is using DistributedOptimizer, it will be similar to DDP class init. otherwise we need to set it manually using hm.setBucketSize().